### ISSUE TITLE
Chooser: hide edit button when user has no permission

### ISSUE THREAD

**Issue by th3hamm0r** [wagtail/wagtail#9039]
Currently, the edit-button for all chooser types is always visible, even if the user has no permission to edit the model. If the user clicks the link, it will redirect him to the home page with a permission error.
This is also the case with older versions of wagtail, but maybe with the new generic choosers in 4.0 (#8480) it could be possible to hide the button accordingly for all types of choosers?

Interestingly, with wagtail 4.0, if you choose an image (probably docs and snippets too), directly after choosing the image, the edit-button is hidden, if the user has no permission for it. After the changes have been saved or the page has been reloaded (if something has been choosen already), the edit-button is visible again.
I assume that this is related to the following view-code for the modal workflow, which uses `AdminURLFinder` passing the actual request-user, when building the data (containing the edit-url) for the frontend:
https://github.com/wagtail/wagtail/blob/df7031f9aeea1a908983c6ca9740afcbccad01c8/wagtail/admin/views/generic/chooser.py#L340-L351

I assume the rendered widget only uses the following code without the request-user, because the request is not available here:
https://github.com/wagtail/wagtail/blob/df7031f9aeea1a908983c6ca9740afcbccad01c8/wagtail/admin/widgets/chooser.py#L196-L206

**Comment by yashp4858** [comment]
@th3hamm0r Is this issue still present?

**Comment by th3hamm0r** [comment]
@yashp4858 yes, this is still present (tested with the current main branch)
